File: mongodb/documentation/source_code/task1/1d.java
/*
private void executeTask1d() {
    Bson filter = or(
            gt("votes.funny", 0),
            gt("votes.useful", 0),
            gt("votes.cool", 0)
    );

    AggregateIterable result = userCollection.aggregate(
            asList(
                    match(filter),
                    group("$name"),
                    sort(ascending("_id"))
            ));

    for (Object o : result) {
        System.out.println(o);
    }
}
*/

File: mongodb/documentation/source_code/task1/1d.js
db.User.aggregate([
    {$match: {$or: [{"votes.funny": {$gt: 0}}, {"votes.useful": {$gt: 0}}, {"votes.cool": {$gt: 0}}]}},
    {$group: {_id: "$name"}},
    {$sort: {"_id": 1}}
]);